addons/im_livechat/models/discuss_channel.py

```diff
@@ -25,6 +25,13 @@ class DiscussChannel(models.Model):
     chatbot_current_step_id = fields.Many2one('chatbot.script.step', string='Chatbot Current Step')
     chatbot_message_ids = fields.One2many('chatbot.message', 'discuss_channel_id', string='Chatbot Messages')
     country_id = fields.Many2one('res.country', string="Country", help="Country of the visitor of the channel")
+    livechat_failure = fields.Selection(
+        selection=[
+            ("never_answered", "Never Answered"),
+            ("no_one_available", "No one Available"),
+        ],
+        string="Live Chat Conversation Failure",
+    )
```

Comment:
If we sometimes want to filter sessions by never_answered or no_one_available we might want on partial index on them two.

Not when it is null, and also probably not when it is no_failure (see other comment) as it will be the big majority of cases so indexing them is useless.